dyn_em/module_first_rk_step_part1.F

```diff
@@ -449,7 +449,7 @@ SUBROUTINE first_rk_step_part1 (   grid , config_flags              &
 
 !gmm halo of wtd and riverflow for leafhydro
 #ifdef DM_PARALLEL
-  IF ( config_flags%sf_surface_physics.eq.NOAHMPSCHEME ) THEN
+  IF ( ( config_flags%sf_surface_physics.eq.NOAHMPSCHEME ) .and. ( grid%STEPWTD .ne. 0 ) )THEN
```

Comment:
@davegill could we instead check (e.g., in check_a_mundo) that STEPWTD > 0?

Reply:
I guess check_a_mundo is not the right place either since this is not coming through the namelist, but is calculated in init

Reply:
Mike,
This is a bug fix, so we have time to get this right for v4.2. If I just turn sf_surface_physics from 2 to 4, with no noahmp namelist record, I get this error. Is there a default value for stepwtd that is better than zero?

Reply:
Dave - 1 is a better default value since this is how often to run groundwater [in timestep units], it should not be <= 0. There is a line in noahmp_init:
STEPWTD = max(STEPWTD,1)
but when groundwater is not active this line is skipped.

Reply:
is this a recent compiler change? I had this happen in some other code recently, i.e., it seems to check all the conditionals instead of checking sequentially.

Reply:
I also don't know why these are implemented as OPTIONAL since they are not packaged separately, but maybe that's an NMM thing.

Reply:
Mike,
I was testing with gnu 9

Reply:
Mike,
Here's the code:
```
       if(iopt_run.eq.5) then
          IF ( PRESENT(smoiseq)     .AND. &
            PRESENT(smcwtdxy)    .AND. &
            PRESENT(rechxy)      .AND. &
            PRESENT(deeprechxy)  .AND. &
            PRESENT(areaxy)      .AND. &
            PRESENT(dx)          .AND. &
            PRESENT(dy)          .AND. &
            PRESENT(msftx)       .AND. &
            PRESENT(msfty)       .AND. &
            PRESENT(wtddt)       .AND. &
            PRESENT(stepwtd)     .AND. &
            PRESENT(dt)          .AND. &
            PRESENT(qrfsxy)      .AND. &
            PRESENT(qspringsxy)  .AND. &
            PRESENT(qslatxy)     .AND. &
            PRESENT(fdepthxy)    .AND. &
            PRESENT(ht)          .AND. &
            PRESENT(riverbedxy)  .AND. &
            PRESENT(eqzwt)       .AND. &
            PRESENT(rivercondxy) .AND. &
            PRESENT(pexpxy)      .AND. &
            PRESENT(rechclim)    ) THEN

             STEPWTD = nint(WTDDT*60./DT)
             STEPWTD = max(STEPWTD,1)

          ELSE
             CALL wrf_error_fatal ('Not enough fields to use groundwater option in Noah-MP')
          END IF
       endif
```

I would have expected stepwtd to be 1 after this. I'll run this down.

Reply:
I think the problem is that this block isn't being called when iopt_run /= 5 so STEPWTD is getting the compiler default initialization, which might be 0.

Reply:
Mike,
Yes, STEPWTD is a state integer, so initialized to zero.

So the logic would be 
```
if iopt_run /= 5 then
   stepwtd = 1
```
?

Reply:
I think what I've done recently in this case is change:
`IF ( config_flags%opt_run.eq.5.and.mod(grid%itimestep,grid%STEPWTD).eq.0 )  THEN`
to:
```
IF ( config_flags%opt_run.eq.5) THEN
  IF (mod(grid%itimestep,grid%STEPWTD).eq.0 )  THEN
```

Reply:
I like your logic, but is there anything special with the OPTIONAL status. I don't think these are actually optional but are declared so.

Reply:
Mike,
Likely OPTIONAL for some NMM reason.

I would like to get this fixed soon so that we can include NoahMP in our testing. What is your preference?

Reply:
To not deal with the optional issue, I prefer this:
```
IF ( config_flags%opt_run.eq.5) THEN
  IF (mod(grid%itimestep,grid%STEPWTD).eq.0 )  THEN
```